src/data/polynomial/eval.lean

```diff
@@ -562,14 +562,25 @@ lemma C_sub : C (a - b) = C a - C b := ring_hom.map_sub C a b
 instance map.is_ring_hom {S} [ring S] (f : R →+* S) : is_ring_hom (map f) :=
 by apply is_ring_hom.of_semiring
 
-@[simp] lemma map_sub {S} [comm_ring S] (f : R →+* S) :
+@[simp] lemma map_sub {S} [ring S] (f : R →+* S) :
   (p - q).map f = p.map f - q.map f :=
 is_ring_hom.map_sub _
 
-@[simp] lemma map_neg {S} [comm_ring S] (f : R →+* S) :
+@[simp] lemma map_neg {S} [ring S] (f : R →+* S) :
   (-p).map f = -(p.map f) :=
 is_ring_hom.map_neg _
 
+@[simp] lemma map_int_cast {S} [ring S] (f : R →+* S) (n : ℤ) :
+  map f ↑n = ↑n :=
```

Comment:
Please use `ring_hom.map_int_cast`.

Reply:
This is the natural thing to do, but `map f` is not a ring homomorphism. We have `map.is_semiring_hom`, but I do not know how to use this.

Reply:
I already did it: we have `ring_hom.of`. BTW, I'm going to make `map` a `ring_hom` but this PR is not ready yet.

Reply:
Oh, nice! I will use this to simplify me next PR.